module/mempool/stdmap/incorporated_result_seals.go

```diff
@@ -200,24 +174,19 @@ func (ir *IncorporatedResultSeals) PruneUpToHeight(height uint64) error {
 			ir.lowestHeight = height
 			return nil
 		}
-		// optimization, if there are less height than the height range to prune,
-		// then just go through each seal.
-		// otherwise, go through each height to prune
+		// Optimization: if there are less seals in the mempool than the height
```

Comment:
more accurately it's comparing with "the height in the index", rather than the total seals, because we are comparing with `len(ir.byHeight)`, not `len(ir.entities)`
```suggestion
		// Optimization: if there are less height in the index than the height range to prune,
```